File: acmp/arch/8.Структуры-данных/0946.Полка/main.py


inlist = [ int(i) for i in open("INPUT.TXT").read().split()[1:] ];
stage = list();
outfile = open("OUTPUT.TXT", "w");

i = 0;
while i < len(inlist):
	action = inlist[i];
	if action == 1:
		stage.insert(0, inlist[i+1]);
		i += 1;
	elif action == 2:
		stage.append(inlist[i+1]);
		i += 1;
	elif action == 3:
		outfile.write( str(stage[0]) + ' ' );
		stage.pop(0);
	else:
		outfile.write( str(stage[-1]) + ' ' );
		stage.pop();
	i += 1;

File: acmp/arch/8.Структуры-данных/0946.Полка/main.cpp
#include <iostream>
#include <fstream>
#include <deque>


using namespace std;





// objects
char const *INPUT_FILE_NAME = "INPUT.TXT";
char const *OUTPUT_FILE_NAME = "OUTPUT.TXT";





// main
int main( int argc, char *argv[] )
{
	int opc;
	deque<int> stage;
	
	
	
	ifstream fin(INPUT_FILE_NAME);
	ofstream fout(OUTPUT_FILE_NAME);

	fin >> opc;
	int action;
	int index;

	while(opc)
	{
		fin >> action;
		switch(action)
		{
		case 1:
			fin >> index;
			stage.push_front(index);
			break;
		case 2:
			fin >> index;
			stage.push_back(index);
			break;
		case 3:
			fout << stage.front() << ' ';
			stage.pop_front();
			break;
		case 4:
			fout << stage.back() << ' ';
			stage.pop_back();
			break;
		}

		--opc;
	}
	

	return 0;
}





// end
